api/client.go

```diff
@@ -121,10 +120,6 @@ func request[Res any](client Client, req *http.Request) (*Res, error) {
 		client.ObserveFunc(start, req, resp, err)
 	}
 
-	if resp != nil && resp.StatusCode == 401 && client.OnUnauthorized != nil {
-		defer client.OnUnauthorized()
```

Comment:
The one difference in this PR is that we no longer `defer` this operation, we do it immediately.

In the connector, cancelling the context immediately seems fine, it does the shutdown a bit earlier.

In the CLI doing the logout operation earlier seems like it should be ok.

Is there something I'm missing that requires these operations to happen after reading the request body?

I guess maybe it'll change the error from 401 to either an error reading the response?

If that's the case I wonder if `ObserveFunc` should return an error, so that we avoid getting to that steps.

Reply:
The only reason `ObserveFunc` is not deferred is so the HTTP client request can be measured without measuring things like reading the request body. It seems one way to resolve this is to take the elapsed time first and pass it into a deferred `ObserveFunc`, e.g. 

```go
if client.ObserveFunc != nil {
    elapsed := time.Since(start)
    defer client.ObserveFunc(elapsed, req, resp, err)
}
```